src/main/scala/model/world/Reproduction.scala

```diff
@@ -24,21 +25,34 @@ object Reproduction {
   /**
    * @param mom a bunny
    * @param dad another bunny
-   * @return the 4 children of the couple, one for each cell of the Punnett's square
+   * @return the 4 children of the couple, one for each cell of the Punnet's square
    */
-  def generateChildren(mom: Bunny, dad: Bunny): Population = {
+  def generateChildren(mom: Bunny, dad: Bunny)(mutations: Option[List[Mutation]]): Population = {
     var childrenGenotypes = List.fill(CHILDREN_EACH_COUPLE)(PartialGenotype(Map()))
+
     Genes.values.foreach(gk => {
       val grandmaMomAllele = mom.genotype(gk).momAllele
       val grandpaMomAllele = mom.genotype(gk).dadAllele
       val grandmaDadAllele = dad.genotype(gk).momAllele
       val grandpaDadAllele = dad.genotype(gk).dadAllele
-      val anotherGene = Random.shuffle(
-        List( Gene(gk, grandmaMomAllele, grandmaDadAllele),
-              Gene(gk, grandpaMomAllele, grandmaDadAllele),
-              Gene(gk, grandmaMomAllele, grandpaDadAllele),
-              Gene(gk, grandpaMomAllele, grandpaDadAllele)))
-      childrenGenotypes = (for (i <- 0 until CHILDREN_EACH_COUPLE) yield childrenGenotypes(i) + anotherGene(i)).toList
+      val genesOfReproduction : List[Gene]= List( Gene(gk, grandmaMomAllele, grandmaDadAllele),
+                                                  Gene(gk, grandpaMomAllele, grandmaDadAllele),
+                                                  Gene(gk, grandmaMomAllele, grandpaDadAllele),
+                                                  Gene(gk, grandpaMomAllele, grandpaDadAllele) )
+      val anotherGene = Random.shuffle(genesOfReproduction)
```

Comment:
Chiamerei direttamente la versione shuffled `genesOfReproduction`

Reply:
quindi intendi rimuovere dalla variabile e richiamare il metodo shuffle() direttamente dove lo usiamo, giusto?

Reply:
Yes

Reply:
Ho provato a chiamare direttamente la posizione da Random.shuffle() per entrambi i casi ma ho il dubbio che ripetendolo nei due case del match possa portare ad avere degli shuffle della lista diversi e quindi magari generare un caso limite in cui casualmente si prende sempre lo stesso gene per cui lascerei il passaggio alla variabile in modo da fare lo shuffle una volta sola

Reply:
Va benissimo, la cosa che mi confondeva in realtà è che prima si chiama "**geneS**ofReproduciton" e poi solo "another**Gene**", metterei gene o al singolare o al plurale in entrambi i casi

Reply:
io metterei entrambi al plurale visto che i geni della riproduzione sono una lista